test/parallel/test-assert-fail.js

```diff
@@ -67,5 +67,5 @@ common.expectsError(() => {
 // The stackFrameFunction should exclude the foo frame
 assert.throws(
   function foo() { assert.fail('first', 'second', 'message', '!==', foo); },
-  (err) => !/foo/m.test(err.stack)
+  (err) => !/foo(?!\/)/m.test(err.stack)
```

Comment:
> So I simply tried to catch for that case, would you happen to know by chance where in that block foo would be mentioned if an actual error went off?

What this part of the test verifies is that the last argument to `assert.fail` in `function foo() { assert.fail('first', 'second', 'message', '!==', foo); },` (the `, foo` one) is not ignored. If you remove that bit, you’ll see something like an additional `at foo` line near the top.

(Since this is obviously not obvious, you could add a comment here that explains what the test does as well?)